axelrod/deterministic_cache.py

```diff
@@ -0,0 +1,158 @@
+try:
+    # Python 2.x
+    from UserDict import UserDict
+except ImportError:
+    # Python 3.x
+    from collections import UserDict
+import dill
+
+from axelrod import Player
+
+
+class DeterministicCache(UserDict):
+    """A class to cache the results of deterministic matches
+
+    For fixed length matches with no noise between pairs of deterministic
```

Comment:
We could make this 'for given length' and add the length of the match to the key. In fact the length should probably be there anyway.